src/components/Search/types.ts

```diff
@@ -130,7 +130,7 @@ type SearchAutocompleteResult = {
 };
 
 type SearchAutocompleteQueryRange = {
-    key: SearchFilterKey;
+    key: SearchFilterKey | typeof CONST.SEARCH.SYNTAX_KEY;
```

Comment:
One suggestion here. There are multiple places in the code where you have to repeat this combined type:
`SearchFilterKey | typeof CONST.SEARCH.SYNTAX_KEY`.
Maybe it would be useful to define
```
type SearchAutocompleteQueryRangeKey = SearchFilterKey | typeof CONST.SEARCH.SYNTAX_KEY
```

and then reuse it in all the query generation functions.